extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSinkBuilder.java

```diff
@@ -0,0 +1,202 @@
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.Serializable;
+
+import static com.hazelcast.jet.impl.util.Util.checkNonNullAndSerializable;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Builder for Elasticsearch Sink
+ * <p>
+ * The Sink first maps items from the pipeline using the provided
+ * {@link #mapItemFn(FunctionEx)} and then using {@link BulkRequest}.
+ * <p>
+ * {@link BulkRequest#BulkRequest()} is used by default, it can be
+ * modified by providing custom {@link #bulkRequestSupplier(SupplierEx)}
+ *
+ * <p>
+ * Usage:
+ * <pre>{@code
+ * Sink<Map<String, ?>> elasticSink = new ElasticSinkBuilder<Map<String, ?>>()
+ *   .clientSupplier(() -> ElasticClients.client(host, port))
+ *   .mapItemFn(item -> new IndexRequest("my-index").source(item))
+ *   .build();
+ * }</pre>
+ * <p>
+ * Requires {@link #clientSupplier(SupplierEx)} and {@link #mapItemFn(FunctionEx)}.
+ *
+ * @param <T>
+ * @since 4.1
+ */
+public class ElasticSinkBuilder<T> implements Serializable {
+
+    private static final String DEFAULT_NAME = "elastic";
+
+    private SupplierEx<? extends RestHighLevelClient> clientSupplier;
+    private ConsumerEx<? super RestHighLevelClient> destroyFn = RestHighLevelClient::close;
+    private SupplierEx<BulkRequest> bulkRequestSupplier = BulkRequest::new;
+    private FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn;
+    private FunctionEx<? super ActionRequest, RequestOptions> optionsFn = (request) -> RequestOptions.DEFAULT;
+    private int preferredLocalParallelism = 2;
+
+    /**
+     * Set the client supplier
+     *
+     * @param clientSupplier supplier for configure Elasticsearch REST client
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> clientSupplier(@Nonnull SupplierEx<? extends RestHighLevelClient> clientSupplier) {
+        this.clientSupplier = checkNonNullAndSerializable(clientSupplier, "clientSupplier");
+        return this;
+    }
+
+    /**
+     * Set the destroy function called on completion, defaults to {@link RestHighLevelClient#close()}
+     *
+     * @param destroyFn destroy function
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> destroyFn(@Nonnull ConsumerEx<? super RestHighLevelClient> destroyFn) {
+        this.destroyFn = checkNonNullAndSerializable(destroyFn, "destroyFn");
+        return this;
+    }
+
+    /**
+     * Set the bulkRequestSupplier, defaults to new {@link BulkRequest#BulkRequest()}
+     *
+     * @param bulkRequestSupplier supplier for the bulk request
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> bulkRequestSupplier(@Nonnull SupplierEx<BulkRequest> bulkRequestSupplier) {
+        this.bulkRequestSupplier = checkNonNullAndSerializable(bulkRequestSupplier, "clientSupplier");
+        return this;
+    }
+
+    /**
+     * Set the mapItemFn
```

Comment:
you can make the javadoc a bit nicer here, for example you can have the main description in the top and then the param can be short, maybe showing an example. We should avoid javadoc like `set X`, and you can show some example code as well.